src/parser/jobs/dnc/modules/FeatherGauge.tsx

```diff
@@ -0,0 +1,116 @@
+import {t} from '@lingui/macro'
+import {Plural, Trans} from '@lingui/react'
+import Color from 'color'
+import _ from 'lodash'
+import React, {Fragment} from 'react'
+
+import TimeLineChart from 'components/ui/TimeLineChart'
+import ACTIONS from 'data/ACTIONS'
+import JOBS from 'data/JOBS'
+import Module, {dependency} from 'parser/core/Module'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+
+import styles from './DNCGauges.module.css'
+
+const FEATHER_GENERATORS = [
+	ACTIONS.REVERSE_CASCADE.id,
+	ACTIONS.FOUNTAINFALL.id,
+	ACTIONS.RISING_WINDMILL.id,
+	ACTIONS.BLOODSHOWER.id,
+]
+
+const FEATHER_CONSUMERS = [
+	ACTIONS.FAN_DANCE.id,
+	ACTIONS.FAN_DANCE_II.id,
+]
+
+const FEATHER_GENERATION_CHANCE = .5
+const MAX_FEATHERS = 4
+
+export default class FeatherGauge extends Module {
+	static handle = 'feathergauge'
+	static title = t('dnc.feather-gauge.title')`Feather Gauge`
+
+	@dependency private suggestions!: Suggestions
+
+	private feathersConsumed = 0
+	private avgGenerated = 0
+	private history: any[] = []
+	private currentFeathers = 0
+	private featherOvercap = 0
+
+	protected init() {
+		this.addHook('cast', {by: 'player', abilityId: FEATHER_GENERATORS}, this.onCastGenerator)
+		this.addHook('cast', {by: 'player', abilityId: FEATHER_CONSUMERS}, this.onConsumeFeather)
+		this.addHook('death', {to: 'player'}, this.onDeath)
+		this.addHook('complete', this.onComplete)
+	}
+	private onCastGenerator() {
+		this.avgGenerated += FEATHER_GENERATION_CHANCE
+		this.setFeather(this.currentFeathers + FEATHER_GENERATION_CHANCE)
+	}
+	private onConsumeFeather() {
+		this.feathersConsumed++
+		// If we consumed a feather when we think we don't have one, clearly we do, so update the history to reflect that
+		// TODO: count how many feathers have been spent since the last possible generation event and endure the history for
```

Comment:
While not quite the same, SAM's kenki module does something similar to this to re-adjust the minimum kenki count on spender usage (keeps the possible range on the graph from spiraling out). Might be able to re-use some of that logic?